src/libraries/System.Memory/tests/ReadOnlySpan/Count.byte.cs

```diff
@@ -0,0 +1,313 @@
+        {
+            ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(Array.Empty<byte>());
+
+            int count = span.Count<byte>(0);
+            Assert.Equal(0, count);
+        }
+        
+        [Fact]
+        public static void ZeroLengthCount_RosByte()
+        {
+            ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(Array.Empty<byte>());
+
+            int count = span.Count<byte>(new ReadOnlySpan<byte>(new byte[] { 0, 0 }));
+            Assert.Equal(0, count);
+        }
+        
+        [Fact]
+        public static void DefaultFilledCount_Byte()
+        {
+            for (int length = 0; length <= byte.MaxValue; length++)
+            {
+                byte[] a = new byte[length];
+                ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(a);
+
+                for (int i = 0; i < length; i++)
+                {
+                    byte target0 = default;
+
+                    int count = span.Count(target0);
+                    Assert.Equal(length,  count);
+                }
+            }
+        }
+        
+        [Fact]
+        public static void DefaultFilledCount_RosByte()
+        {
+            for (int length = 0; length <= byte.MaxValue; length++)
+            {
+                byte[] a = new byte[length];
+                ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(a);
+
+                for (int i = 0; i < length; i++)
+                {
+                    ReadOnlySpan<byte> target0 = new byte[] { 0, 0 };
+
+                    int count = span.Count(target0);
+                    Assert.Equal(length / 2,  count);
+                }
+            }
+        }
+        
+        [Fact]
+        public static void TestCount_Byte()
+        {
+            for (int length = 0; length <= byte.MaxValue; length++)
+            {
+                byte[] a = new byte[length];
+                for (int i = 0; i < length; i++)
+                {
+                    a[i] = (byte)(i + 1);
+                }
+                ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(a);
+
+                for (int targetIndex = 0; targetIndex < length; targetIndex++)
+                {
+                    byte target = a[targetIndex];
+
+                    int count = span.Count(target);
+                    Assert.Equal(1, count);
+                }
+            }
+        }
+        
+        [Fact]
+        public static void TestCount_RosByte()
+        {
+            for (int length = 0; length <= byte.MaxValue; length++)
+            {
+                byte[] a = new byte[length];
+                for (int i = 0; i < length; i++)
+                {
+                    a[i] = (byte)(i + 1);
+                }
+                ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(a);
+
+                for (int targetIndex = 0; targetIndex < length - 1; targetIndex++)
+                {
+                    ReadOnlySpan<byte> target = stackalloc byte[] { a[targetIndex], a[targetIndex + 1] };
```

Comment:
stackalloc within a loop is dangerous and in a test like this we can just use new arrays...
```suggestion
                    ReadOnlySpan<byte> target = new byte[] { a[targetIndex], a[targetIndex + 1] };
```
Alternatively you could create the array once before the loop and then just repopulate it on each call